src/api/app/views/webui/users/subscriptions/index.js.erb

```diff
@@ -0,0 +1,2 @@
+$('#subscriptions-form').replaceWith('<%= escape_javascript(render(partial: 'webui/subscriptions/subscriptions_form')) %>');
```

Comment:
This seems like quite a round-about way of doing implementing this feature, I mean, we have many steps to implement this feature:
1. Click Reset link 
2. Send ajax requests to SubscriptionsController#index
3. Return javascript to replace #subscriptions-form with the default subscriptions form html rendered in the javascript.

Why not just make the Reset link be a normal request to the controller? Also FYI reseting a user's subscriptions to the default can also be achieved by deleting all of the user's subscriptions. So it could be done with a destroy method on the controller i.e.
```ruby
  def destroy
    subscriptions_form.reset_to_default
  end
```
And then in event_subscription/form.rb:
```ruby
    def reset_to_default
      subscriber.event_subscriptions.delete_all
    end
```

Reply:
@evanrolfe With the current approach a user can see the change and then decide if he wants it or not. Your proposal would directly apply the changes.